### ISSUE TITLE
Terraform fails to retrieve schema.Set resource attribute value containing references to other resources.

### ISSUE THREAD

**Issue by ggiamarchi** [hashicorp/terraform#873]
For instance, using the sample provider [provider.go](https://gist.github.com/ggiamarchi/cc6f200842794c0b05eb#file-provider-go) and the corresponding terraform configuration :

```
provider "test" {
}

resource "test_container" "ctn001" {
    elements = [
        "${test_element.elt001.id}",
        "${test_element.elt002.id}",
        "${test_element.elt003.id}"
    ]
}

resource "test_element" "elt001" {
}

resource "test_element" "elt002" {
}

resource "test_element" "elt003" {
}
```
Terraform creates the 3 `test_element` resources but then, when it tries to create the `test_container` resource it doesn't give the right IDs when getting it with [`d.Get("elements").(*schema.Set)`](https://gist.github.com/ggiamarchi/cc6f200842794c0b05eb#file-provider-go-L35)

In the log, we can see the `elements` set is wrong

```
2015/01/28 02:51:13 terraform-provider-test: 2015/01/28 02:51:13 [DEBUG] Create element with id XVlBzgbaiCMRAjWwhTHctcuAxhxKQF
2015/01/28 02:51:13 terraform-provider-test: 2015/01/28 02:51:13 [DEBUG] Create element with id DaFpLSjFbcXoEFfRsWxPLDnJObCsNV
2015/01/28 02:51:13 terraform-provider-test: 2015/01/28 02:51:13 [DEBUG] Create element with id lgTeMaPEZQleQYhYzRyWJjPjzpfRFE
...
2015/01/28 02:51:13 terraform-provider-test: 2015/01/28 02:51:13 [DEBUG] Create container with id gmotaFetHsbZRjxAwnwekrBEmfdzdc and elements *Set(map[int]interface {}{616279343:"XVlBzgbaiCMRAjWwhTHctcuAxhxKQF", 0:"", 471749553:"DaFpLSjFbcXoEFfRsWxPLDnJObCsNV"})
test_element.elt003: Creation complete
...
test_element.elt002: Creation complete
test_element.elt001: Creation complete
test_container.ctn001: Creating...
  elements.#:           "0" => "3"
  elements.471749553:   "" => "DaFpLSjFbcXoEFfRsWxPLDnJObCsNV"
  elements.616279343:   "" => "XVlBzgbaiCMRAjWwhTHctcuAxhxKQF"
  elements.~1468225307: "" => ""
test_container.ctn001: Creation complete
```

This is a regression introduced by https://github.com/hashicorp/terraform/commit/b4bf8131516c80a0b3b980924afde5d510458935. When testing with the commit just before https://github.com/hashicorp/terraform/commit/aa82c10e3927eb33d5ecbc5844e93619a088d638 it works great as shown in the following log

```
2015/01/28 02:57:45 terraform-provider-test: 2015/01/28 02:57:45 [DEBUG] Create element with id XVlBzgbaiCMRAjWwhTHctcuAxhxKQF
2015/01/28 02:57:45 terraform-provider-test: 2015/01/28 02:57:45 [DEBUG] Create element with id DaFpLSjFbcXoEFfRsWxPLDnJObCsNV
2015/01/28 02:57:45 terraform-provider-test: 2015/01/28 02:57:45 [DEBUG] Create element with id lgTeMaPEZQleQYhYzRyWJjPjzpfRFE
...
test_element.elt003: Creation complete
test_element.elt002: Creation complete
test_element.elt001: Creation complete
test_container.ctn001: Creating...
  elements.#:           "0" => "3"
  elements.471749553:   "" => "DaFpLSjFbcXoEFfRsWxPLDnJObCsNV"
  elements.616279343:   "" => "XVlBzgbaiCMRAjWwhTHctcuAxhxKQF"
  elements.~1468225307: "" => "lgTeMaPEZQleQYhYzRyWJjPjzpfRFE"
test_container.ctn001: Creation complete
```

**Comment by mitchellh** [comment]
I haven't been able to repro this so far, but watched it repro on someone else's. I thought maybe race condition and made this script, but nothing so far:

```
echo Starting...
count=1
while [ $? -eq 0 ]; do
    rm -f terraform.tfstate
    output=$(terraform apply)
    if [[ $output == *"~"* ]]; then
        echo "Reproduced on count ${count}!"
        echo $output
        exit 1
    fi

    ((count++))
    if [ $count -gt 100 ]; then
        echo No bug encountered.
        exit 0
    fi
done
```

**Comment by ggiamarchi** [comment]
Here is the `terraform plan` output

```console
$ terraform plan

2015/01/31 14:52:57 [INFO] Terraform version: 0.3.7 dev 170ddc4bd48d4ab854ab458e1446180ccdb88095+CHANGES
2015/01/31 14:52:57 Detected home directory from env var: /Users/guillaume
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: atlas = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-atlas
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: aws = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-aws
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: azure = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-azure
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: cloudflare = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-cloudflare
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: cloudstack = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-cloudstack
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: consul = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-consul
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: digitalocean = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-digitalocean
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: dnsimple = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-dnsimple
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: google = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-google
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: heroku = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-heroku
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: mailgun = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-mailgun
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: null = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-null
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: test = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: file = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provisioner-file
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: local-exec = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provisioner-local-exec
2015/01/31 14:52:57 [DEBUG] Discoverd plugin: remote-exec = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provisioner-remote-exec
2015/01/31 14:52:57 Detected home directory from env var: /Users/guillaume
2015/01/31 14:52:57 [DEBUG] Attempting to open CLI config file: /Users/guillaume/.terraformrc
2015/01/31 14:52:57 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2015/01/31 14:52:57 [DEBUG] Creating graph for path: [root]
2015/01/31 14:52:57 [DEBUG] Starting plugin: /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test []string{"/Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test"}
2015/01/31 14:52:57 [DEBUG] Waiting for RPC address for: /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test
2015/01/31 14:52:57 Detected home directory from env var: /Users/guillaume
2015/01/31 14:52:57 terraform-provider-test: 2015/01/31 14:52:57 Plugin address: unix /var/folders/6p/3fj1m23j24lbd4x7brp7dfgw0000gn/T/tf-plugin108791787
2015/01/31 14:52:57 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:52:57 [INFO] Validating provider: test
2015/01/31 14:52:57 [DEBUG] test_element.elt003: expanding to count = 1
2015/01/31 14:52:57 [DEBUG] test_element.elt002: expanding to count = 1
2015/01/31 14:52:57 [DEBUG] test_element.elt001: expanding to count = 1
2015/01/31 14:52:57 [INFO] Validating resource: test_element.elt003
2015/01/31 14:52:57 [INFO] Validating resource: test_element.elt002
2015/01/31 14:52:57 [INFO] Validating resource: test_element.elt001
2015/01/31 14:52:57 [DEBUG] test_container.ctn001: expanding to count = 1
2015/01/31 14:52:57 [INFO] Validating resource: test_container.ctn001
2015/01/31 14:52:57 [DEBUG] Creating graph for path: [root]
2015/01/31 14:52:57 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:52:57 [DEBUG] Creating graph for path: [root]
Refreshing Terraform state prior to plan...

2015/01/31 14:52:57 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:52:57 [INFO] Configuring provider: test
2015/01/31 14:52:57 [DEBUG] test_element.elt003: expanding to count = 1
2015/01/31 14:52:57 [DEBUG] test_element.elt002: expanding to count = 1
2015/01/31 14:52:57 [DEBUG] test_element.elt001: expanding to count = 1
2015/01/31 14:52:57 [INFO] Module root walking: test_element.elt003 (Graph node: test_element.elt003)
2015/01/31 14:52:57 [DEBUG] test_element.elt003: Not refreshing, ID is empty
2015/01/31 14:52:57 [INFO] Module root walking: test_element.elt002 (Graph node: test_element.elt002)
2015/01/31 14:52:57 [DEBUG] test_element.elt002: Not refreshing, ID is empty
2015/01/31 14:52:57 [INFO] Module root walking: test_element.elt001 (Graph node: test_element.elt001)
2015/01/31 14:52:57 [DEBUG] test_element.elt001: Not refreshing, ID is empty
2015/01/31 14:52:57 [DEBUG] test_container.ctn001: expanding to count = 1
2015/01/31 14:52:57 [INFO] Module root walking: test_container.ctn001 (Graph node: test_container.ctn001)
2015/01/31 14:52:57 [DEBUG] test_container.ctn001: Not refreshing, ID is empty
2015/01/31 14:52:57 [DEBUG] Creating graph for path: [root]

2015/01/31 14:52:57 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:52:57 [INFO] Configuring provider: test
2015/01/31 14:52:57 [DEBUG] test_element.elt003: expanding to count = 1
2015/01/31 14:52:57 [DEBUG] test_element.elt002: expanding to count = 1
2015/01/31 14:52:57 [DEBUG] test_element.elt001: expanding to count = 1
2015/01/31 14:52:57 [INFO] Module root walking: test_element.elt003 (Graph node: test_element.elt003)
2015/01/31 14:52:57 [DEBUG] test_element.elt003: Executing diff
2015/01/31 14:52:57 [INFO] Module root walking: test_element.elt002 (Graph node: test_element.elt002)
2015/01/31 14:52:57 [DEBUG] test_element.elt002: Executing diff
2015/01/31 14:52:57 [INFO] Module root walking: test_element.elt001 (Graph node: test_element.elt001)
2015/01/31 14:52:57 [DEBUG] test_element.elt001: Executing diff
2015/01/31 14:52:57 [DEBUG] test_element.elt003: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}}, Destroy:false, DestroyTainted:false}
2015/01/31 14:52:57 [DEBUG] test_element.elt002: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}}, Destroy:false, DestroyTainted:false}
2015/01/31 14:52:57 [DEBUG] test_element.elt001: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}}, Destroy:false, DestroyTainted:false}
2015/01/31 14:52:57 [DEBUG] test_container.ctn001: expanding to count = 1
2015/01/31 14:52:57 [INFO] Module root walking: test_container.ctn001 (Graph node: test_container.ctn001)
2015/01/31 14:52:57 [DEBUG] test_container.ctn001: Executing diff
2015/01/31 14:52:57 [DEBUG] test_container.ctn001: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"elements.#":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}}, Destroy:false, DestroyTainted:false}
The Terraform execution plan has been generated and is shown below.
2015/01/31 14:52:57 waiting for all plugin processes to complete...
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed.

2015/01/31 14:52:57 [DEBUG] /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test: plugin process exited
Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

+ test_container.ctn001
    elements.#: "" => "<computed>"

+ test_element.elt001

+ test_element.elt002

+ test_element.elt003
```

**Comment by ggiamarchi** [comment]
And the `apply` full output

```
2015/01/31 14:55:55 [INFO] Terraform version: 0.3.7 dev 170ddc4bd48d4ab854ab458e1446180ccdb88095+CHANGES
2015/01/31 14:55:55 Detected home directory from env var: /Users/guillaume
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: atlas = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-atlas
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: aws = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-aws
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: azure = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-azure
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: cloudflare = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-cloudflare
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: cloudstack = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-cloudstack
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: consul = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-consul
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: digitalocean = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-digitalocean
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: dnsimple = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-dnsimple
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: google = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-google
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: heroku = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-heroku
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: mailgun = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-mailgun
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: null = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-null
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: test = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: file = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provisioner-file
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: local-exec = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provisioner-local-exec
2015/01/31 14:55:55 [DEBUG] Discoverd plugin: remote-exec = /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provisioner-remote-exec
2015/01/31 14:55:55 Detected home directory from env var: /Users/guillaume
2015/01/31 14:55:55 [DEBUG] Attempting to open CLI config file: /Users/guillaume/.terraformrc
2015/01/31 14:55:55 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2015/01/31 14:55:55 [DEBUG] Creating graph for path: [root]
2015/01/31 14:55:55 [DEBUG] Starting plugin: /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test []string{"/Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test"}
2015/01/31 14:55:55 [DEBUG] Waiting for RPC address for: /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test
2015/01/31 14:55:55 Detected home directory from env var: /Users/guillaume
2015/01/31 14:55:55 terraform-provider-test: 2015/01/31 14:55:55 Plugin address: unix /var/folders/6p/3fj1m23j24lbd4x7brp7dfgw0000gn/T/tf-plugin024505988
2015/01/31 14:55:55 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:55:55 [DEBUG] Creating graph for path: [root]
2015/01/31 14:55:55 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:55:55 [INFO] Validating provider: test
2015/01/31 14:55:55 [DEBUG] test_element.elt003: expanding to count = 1
2015/01/31 14:55:55 [DEBUG] test_element.elt002: expanding to count = 1
2015/01/31 14:55:55 [DEBUG] test_element.elt001: expanding to count = 1
2015/01/31 14:55:55 [INFO] Validating resource: test_element.elt003
2015/01/31 14:55:55 [INFO] Validating resource: test_element.elt002
2015/01/31 14:55:55 [INFO] Validating resource: test_element.elt001
2015/01/31 14:55:55 [DEBUG] test_container.ctn001: expanding to count = 1
2015/01/31 14:55:55 [INFO] Validating resource: test_container.ctn001
2015/01/31 14:55:55 [DEBUG] Creating graph for path: [root]
2015/01/31 14:55:55 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:55:55 [INFO] Configuring provider: test
2015/01/31 14:55:55 [DEBUG] test_element.elt003: expanding to count = 1
2015/01/31 14:55:55 [DEBUG] test_element.elt002: expanding to count = 1
2015/01/31 14:55:55 [DEBUG] test_element.elt001: expanding to count = 1
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt003 (Graph node: test_element.elt003)
2015/01/31 14:55:55 [DEBUG] test_element.elt003: Not refreshing, ID is empty
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt002 (Graph node: test_element.elt002)
2015/01/31 14:55:55 [DEBUG] test_element.elt002: Not refreshing, ID is empty
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt001 (Graph node: test_element.elt001)
2015/01/31 14:55:55 [DEBUG] test_element.elt001: Not refreshing, ID is empty
2015/01/31 14:55:55 [DEBUG] test_container.ctn001: expanding to count = 1
2015/01/31 14:55:55 [INFO] Module root walking: test_container.ctn001 (Graph node: test_container.ctn001)
2015/01/31 14:55:55 [DEBUG] test_container.ctn001: Not refreshing, ID is empty
2015/01/31 14:55:55 [DEBUG] Creating graph for path: [root]
2015/01/31 14:55:55 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:55:55 [INFO] Configuring provider: test
2015/01/31 14:55:55 [DEBUG] test_element.elt003: expanding to count = 1
2015/01/31 14:55:55 [DEBUG] test_element.elt002: expanding to count = 1
2015/01/31 14:55:55 [DEBUG] test_element.elt001: expanding to count = 1
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt003 (Graph node: test_element.elt003)
2015/01/31 14:55:55 [DEBUG] test_element.elt003: Executing diff
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt002 (Graph node: test_element.elt002)
2015/01/31 14:55:55 [DEBUG] test_element.elt002: Executing diff
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt001 (Graph node: test_element.elt001)
2015/01/31 14:55:55 [DEBUG] test_element.elt001: Executing diff
2015/01/31 14:55:55 [DEBUG] test_element.elt003: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}}, Destroy:false, DestroyTainted:false}
2015/01/31 14:55:55 [DEBUG] test_element.elt002: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}}, Destroy:false, DestroyTainted:false}
2015/01/31 14:55:55 [DEBUG] test_element.elt001: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}}, Destroy:false, DestroyTainted:false}
2015/01/31 14:55:55 [DEBUG] test_container.ctn001: expanding to count = 1
2015/01/31 14:55:55 [INFO] Module root walking: test_container.ctn001 (Graph node: test_container.ctn001)
2015/01/31 14:55:55 [DEBUG] test_container.ctn001: Executing diff
2015/01/31 14:55:55 [DEBUG] test_container.ctn001: Diff: &terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"elements.#":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}, "id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:true, Type:0x2}}, Destroy:false, DestroyTainted:false}
2015/01/31 14:55:55 [INFO] Apply walk starting
2015/01/31 14:55:55 [DEBUG] Creating graph for path: [root]
2015/01/31 14:55:55 [DEBUG] Graph [root] created and valid. 6 nouns.
2015/01/31 14:55:55 [INFO] Configuring provider: test
2015/01/31 14:55:55 [DEBUG] test_element.elt003: expanding to count = 1
2015/01/31 14:55:55 [DEBUG] test_element.elt002: expanding to count = 1
2015/01/31 14:55:55 [DEBUG] test_element.elt001: expanding to count = 1
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt003 (Graph node: test_element.elt003)
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt002 (Graph node: test_element.elt002)
2015/01/31 14:55:55 [INFO] Module root walking: test_element.elt001 (Graph node: test_element.elt001)
2015/01/31 14:55:55 [DEBUG] test_element.elt003: Executing Apply
test_element.elt003: Creating...
2015/01/31 14:55:55 [DEBUG] test_element.elt002: Executing Apply
test_element.elt002: Creating...
test_element.elt001: Creating...
2015/01/31 14:55:55 [DEBUG] test_element.elt001: Executing Apply
2015/01/31 14:55:55 terraform-provider-test: 2015/01/31 14:55:55 [DEBUG] Create element with id XVlBzgbaiCMRAjWwhTHctcuAxhxKQF
2015/01/31 14:55:55 terraform-provider-test: 2015/01/31 14:55:55 [DEBUG] Create element with id DaFpLSjFbcXoEFfRsWxPLDnJObCsNV
2015/01/31 14:55:55 terraform-provider-test: 2015/01/31 14:55:55 [DEBUG] Create element with id lgTeMaPEZQleQYhYzRyWJjPjzpfRFE
test_element.elt003: Creation complete
2015/01/31 14:55:55 [DEBUG] test_container.ctn001: expanding to count = 1
2015/01/31 14:55:55 [INFO] Module root walking: test_container.ctn001 (Graph node: test_container.ctn001)
test_element.elt002: Creation complete
test_element.elt001: Creation complete
2015/01/31 14:55:55 [DEBUG] test_container.ctn001: Executing Apply
test_container.ctn001: Creating...
  elements.#:           "0" => "3"
  elements.471749553:   "" => "DaFpLSjFbcXoEFfRsWxPLDnJObCsNV"
2015/01/31 14:55:55 terraform-provider-test: 2015/01/31 14:55:55 [DEBUG] Create container with id gmotaFetHsbZRjxAwnwekrBEmfdzdc and elements *Set(map[int]interface {}{0:"", 471749553:"DaFpLSjFbcXoEFfRsWxPLDnJObCsNV", 616279343:"XVlBzgbaiCMRAjWwhTHctcuAxhxKQF"})
  elements.616279343:   "" => "XVlBzgbaiCMRAjWwhTHctcuAxhxKQF"
2015/01/31 14:55:55 [INFO] Apply walk complete
2015/01/31 14:55:55 [INFO] Writing backup state to: terraform.tfstate.backup
  elements.~1468225307: "" => ""
test_container.ctn001: Creation complete

Apply complete! Resources: 4 added, 0 changed, 0 destroyed.
2015/01/31 14:55:55 waiting for all plugin processes to complete...

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

2015/01/31 14:55:55 [DEBUG] /Users/guillaume/dev/go/src/github.com/hashicorp/terraform/bin/terraform-provider-test: plugin process exited
State path: terraform.tfstate
```

**Comment by mitchellh** [comment]
Fixed by #967 apparently!
